apps/studio/src/lib/db/clients/sqlserver.ts

```diff
@@ -914,23 +915,18 @@ export class SQLServerClient extends BasicDatabaseClient<SQLServerResult> {
     this.logger().info('RUNNING', queryArgs)
 
     const runQuery = async (connection) => {
-      const request = connection.request()
-      request.arrayRowMode = arrayRowMode
-      const data = await request.query(queryArgs.query)
+      connection.arrayRowMode = arrayRowMode
+      const data = await connection.query(queryArgs.query)
       const rowsAffected = _.sum(data.rowsAffected)
-      return { request, data, rowsAffected }
+      return { connection, data, rowsAffected }
     };
 
-    return this.connection
-      ? runQuery(this.connection)
-      : this.runWithConnection(runQuery)
+    return this.runWithConnection(runQuery)
   }
 
   private async runWithConnection(run) {
-    const connection = await new ConnectionPool(this.dbConfig).connect()
-    this.connection = connection
-    this.connection.dbConfig = this.dbConfig
-    return run(this.connection)
+    const connection = this.pool.request();
```

Comment:
@not-night-but don't you need to `release` the connection after calling `run`?

Reply:
I don't think so?

![image](https://github.com/beekeeper-studio/beekeeper-studio/assets/58712803/40474d06-e9b0-4738-a50c-88a529c92d1e)

Reply:
Huh, that's really weird

Reply:
Is `Request` the right thing here?

Reply:
I believe that's what we were using before, but I'll see if there's something more idiomatic